src/doom/r_main.c

```diff
@@ -699,16 +700,36 @@ void R_InitLightTables (void)
    // [crispy] smooth diminishing lighting
     if (crispy->smoothlight)
     {
-	LIGHTLEVELS = 32;
-	LIGHTSEGSHIFT = 3;
-	LIGHTBRIGHT = 2;
-	MAXLIGHTSCALE = 48;
-	LIGHTSCALESHIFT = 12;
-	MAXLIGHTZ = 1024;
-	LIGHTZSHIFT = 17;
+#ifdef CRISPY_TRUECOLOR
+    if (crispy->truecolor)
+    {
+	    // [crispy] if in TrueColor mode, use smoothest diminished lighting
+	    NUMCOLORMAPS = 256;
```

Comment:
Could you formulate the values as multiples of the Vanilla values, e.g. `LIGHTLEVELS = 16 * 16`, and the shifts as e.g. `LIGHTSEGSHIFT = 4 - 4`?

Reply:
Sure, here it is:
```
	    // [crispy] if in TrueColor mode, use smoothest diminished lighting
	    NUMCOLORMAPS = 32 * 8;      // 256
	    LIGHTLEVELS = 16 * 16;      // 256
	    LIGHTSEGSHIFT = 4 - 4;      // 0
	    LIGHTBRIGHT = 1 + 14;       // 15
	    MAXLIGHTSCALE = 48 * 8;     // 384, but initial is 376 / 48 = 7,833333333333333
	    LIGHTSCALESHIFT = 12 - 3;   // 9
	    MAXLIGHTZ = 128 * 8;        // 1024
	    LIGHTZSHIFT = 20 - 3;       // 17
```

Reply:
But if it's not a secret, what these multipliers and substractors can give? They are _mostly_ different from each other.

Reply:
Take for example `MAXLIGHTZ`. You increase its value by factor `8`, so this means we have `8` times as many light levels. To map distance to light levels, the distance value is right-shifted by the value of `LIGHTZSHIFT`. You'll have to decrease this value by `3`, because `8 = 2^3 = 1 << 3`, to map the distance value into a light level range that is `8` times as large. All values are connected with each other in one way or another. 😉

Reply:
> that 1% inaccuracy is simply impossible to see while normal playing and not comparing screenshots. The game still looks as it should, though soft diminishing is more noticeable in darker areas. I think it's all absolutely fine, at least in visual aspect.

That's what I say! 😉